integration/data/test_restore_to.py

```diff
@@ -0,0 +1,252 @@
+import os
+import pyqcow
+
+import cmd
+import common
+from common import replica1, replica2, controller, backing_replica1, backing_replica2, backup_targets  # NOQA
+from common import read_dev, read_file, read_from_backing_file
+from utils import file, create_backup, rm_backups, rm_snaps
+
+VOLUME_NAME = 'test-volume_1.0'
+VOLUME_SIZE = str(4 * 1024 * 1024)  # 4M
+BLOCK_SIZE = str(2 * 1024 * 1024)  # 2M
+
+BACKING_FILE_QCOW2 = 'backing_file.qcow2'
+
+OUTPUT_FILE_RAW = 'test-restore_to-output.raw'
+OUTPUT_FILE_QCOW2 = 'test-restore_to-output.qcow2'
+
+IMAGE_FORMAT_RAW = 'raw'
+IMAGE_FORMAT_QCOW2 = 'qcow2'
+
+
+def check_backing(dev, offset=0, length=4*1024):
+    backing_data = read_from_backing_file(offset, length)
+    empty_volume_data = read_dev(dev, offset, length)
+    assert backing_data == empty_volume_data
+
+
+def read_qcow2_file_without_backing_file(file, offset, length):
```

Comment:
what do you mean by `without_backing_file` here? I didn't see a similar function `with_backing_file`.